src/main/java/org/broadinstitute/hellbender/tools/sv/cluster/CanonicalSVCollapser.java

```diff
@@ -0,0 +1,554 @@
+package org.broadinstitute.hellbender.tools.sv.cluster;
+
+import com.google.common.annotations.VisibleForTesting;
+import htsjdk.samtools.reference.ReferenceSequenceFile;
+import htsjdk.variant.variantcontext.*;
+import org.apache.commons.lang3.tuple.Pair;
+import org.apache.commons.math3.stat.descriptive.rank.Percentile;
+import org.broadinstitute.hellbender.tools.spark.sv.utils.GATKSVVCFConstants;
+import org.broadinstitute.hellbender.tools.sv.SVCallRecord;
+import org.broadinstitute.hellbender.tools.sv.SVCallRecordUtils;
+import org.broadinstitute.hellbender.utils.MathUtils;
+import org.broadinstitute.hellbender.utils.Utils;
+import org.broadinstitute.hellbender.utils.reference.ReferenceUtils;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+/**
+ * Class for collapsing a collection of similar {@link SVCallRecord} objects, such as clusters produced by
+ * {@link CanonicalSVLinkage}, into a single representative call.
+ */
+public class CanonicalSVCollapser implements SVCollapser<SVCallRecord> {
+
+    /**
+     * Determines whether a given SV type represents a CNV.
+     */
+    public static boolean isCnvType(final StructuralVariantType type) {
+        return type == StructuralVariantType.DEL || type == StructuralVariantType.DUP || type == StructuralVariantType.CNV;
+    }
+
+    /**
+     * Define strategies for collapsing variant intervals.
+     */
+    public enum BreakpointSummaryStrategy {
```

Comment:
I think I see what you're saying here - if `CanonicalSVCollapser` has a subclass that redefines `BreakpointSummaryStrategy` it may get confusing. I just tested this and it is allowed to define an `BreakpointSummaryStrategy` enum type in a subclass, but the `enum`'s will be distinct - i.e. `SVCollapserSubclass.BreakpointSummaryStrategy` is not the same as `CanonicalSVCollapser.BreakpointSummaryStrategy`. In that case we would probably just use a different name and ensure the subclass uses the proper `BreakpointSummaryStrategy` if/where needed.

The design isn't ideal for subclasses, but I'd be okay with leaving refactoring til later if we do need to do that.